app/models/bulkrax/csv_entry.rb

```diff
@@ -90,6 +86,14 @@ def build_metadata
       self.parsed_metadata
     end
 
+    def validate_record
+      raise StandardError, 'Record not found' if record.nil?
+      unless importerexporter.parser.required_elements?(record)
+        raise StandardError,
+              "Missing required elements, missing element(s) are: #{importerexporter.parser.missing_elements(keys_without_numbers(keys_with_values)).join(', ')}"
```

Comment:
One adjustment is that the `required_elements?` now knows what the keys are that are missing.  And there's a duplication of that knowledge here.  But I definitely like this pathway.